### PR TITLE
SDKS-2251-Webauthn key selector sort

### PR DESCRIPTION
**Pull request by jeyanthanperiyasamy**
# JIRA Ticket

Please, link jira ticket here.

# Description

Briefly describe the change and any information that would help speedup the review and testing process.

# Definition of Done Checklist:

- [ ] Acceptance criteria is met.
- [ ] All tasks listed in the user story have been completed.
- [ ] Coded to standards.
- [ ] Ensure backward compatibility.
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [ ] e2e tests are written.
- [ ] Functional spec is written/updated.
- [ ] Example code snippets have been added.
- [ ] Change log updated.
- [ ] Documentation story is created and tracked.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

### CODE REVIEW

**Review by george-bafaloukas-forgerock — APPROVED** [review]
LGTM

File: FRAuth/FRAuth/WebAuthn/Authenticator/PlatformAuthenticator/PlatformAuthenticatorGetAssertionSession.swift [hunk 1]
```diff
@@ -357,3 +358,21 @@ class PlatformAuthenticatorGetAssertionSession: AuthenticatorGetAssertionSession
...
+            let value: [String] = key.components(separatedBy: " ")
+            return (key, value.count > 1 ? formatter.date(from: value[1]) : nil)
+        }.sorted { lhs, rhs in
+            if let lhsDate = lhs.1 , let rhsDate = rhs.1 {
+                return lhsDate > rhsDate
+            }
+            return lhs.0 > rhs.0
+        }.map { $0.0 }
+    }
+}
```
Comment (vahancouver) [inline FRAuth/FRAuth/WebAuthn/Authenticator/PlatformAuthenticator/PlatformAuthenticatorGetAssertionSession.swift:377]: I think the name should be **sortedByDate** as it is a not-mutating method.

I'm suggesting this version of the method that is more readable and doesn't use the extra maps:

```
func sortedByDate() -> [String] {
        let formatter = DateFormatter()
        formatter.dateFormat = "yyyyMMdd HH:mm:ss"
        
        return self.sorted { string1, string2 in
            let stringComponents1 = string1.components(separatedBy: " ")
            let stringComponents2 = string2.components(separatedBy: " ")
            
            let date1 = stringComponents1.count > 1 ? formatter.date(from: stringComponents1[1]) : nil
            let date2 = stringComponents2.count > 1 ? formatter.date(from: stringComponents2[1]) : nil
            
            if let date1 = date1, let date2 = date2 {
                return date1 > date2
            }
            return string1 > string2
        }
    }
```

Comment (jeyanthanperiyasamy) [reply FRAuth/FRAuth/WebAuthn/Authenticator/PlatformAuthenticator/PlatformAuthenticatorGetAssertionSession.swift:377]: With Map

.periyasamy/Projects/iOS/forgerock-ios-sdk/FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/WebAuthn/WebAuthnKeySelectorTests.swift:55: Test Case '-[FRAuthTests.WebAuthnKeySelectorTests testKeysWithNullOrNoDate]' measured [Time, seconds] average: 0.000, relative standard deviation: 173.540%, values: [0.002351, 0.000192, 0.000165, 0.000157, 0.000154, 0.000152, 0.000151, 0.000161, 0.000151, 0.000154], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , polarity: prefers smaller, maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100
Test Case '-[FRAuthTests.WebAuthnKeySelectorTests testKeysWithNullOrNoDate]' passed (0.275 seconds).



Without Map

/Users/jey.periyasamy/Projects/iOS/forgerock-ios-sdk/FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/WebAuthn/WebAuthnKeySelectorTests.swift:55: Test Case '-[FRAuthTests.WebAuthnKeySelectorTests testKeysWithNullOrNoDate]' measured [Time, seconds] average: 0.001, relative standard deviation: 170.804%, values: [0.005094, 0.000724, 0.000600, 0.000497, 0.000245, 0.000242, 0.000242, 0.000242, 0.000241, 0.000241], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , polarity: prefers smaller, maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100

Comment (jeyanthanperiyasamy) [reply FRAuth/FRAuth/WebAuthn/Authenticator/PlatformAuthenticator/PlatformAuthenticatorGetAssertionSession.swift:377]: Both are closer for the average standard deviation . but i like vahan approach for the space complexity

**Review by vahancouver — APPROVED** [review]
LGTM 👍

**Review by vahancouver — APPROVED** [review]
LGTM

**Review by spetrov — APPROVED** [review]

**Pull request merged**